src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -261,12 +266,17 @@ public DynamicObject multiply(DynamicObject string, int times,
             return allocateObjectNode.allocate(Layouts.BASIC_OBJECT.getLogicalClass(string), Layouts.STRING.build(false, false, repeated));
         }
 
+        @Specialization(guards = "times >= 0")
```

Comment:
You need to check for `!fitsInInteger` here, otherwise once this specialization is installed it would take every positive value, including the ones in int range.